ydb/docs/en/core/dev/example-app/_includes/example-cpp.md

```diff
@@ -206,174 +185,213 @@ The given code snippet prints the following text to the console at startup:
 
 The first step is to prepare and execute the first query:
 
 ```c++
 //! Shows usage of transactions consisting of multiple data queries with client logic between them.
-static TStatus MultiStepTransaction(TSession session, const TString& path, ui64 seriesId, ui64 seasonId,
-    TMaybe<TResultSet>& resultSet)
-{
-    auto query1 = Sprintf(R"(
-        PRAGMA TablePathPrefix("%s");
-
-        DECLARE $seriesId AS Uint64;
-        DECLARE $seasonId AS Uint64;
-
-        SELECT first_aired AS from_date FROM seasons
-        WHERE series_id = $seriesId AND season_id = $seasonId;
-    )", path.c_str());
-
-    auto params1 = session.GetParamsBuilder()
-        .AddParam("$seriesId")
-            .Uint64(seriesId)
-            .Build()
-        .AddParam("$seasonId")
-            .Uint64(seasonId)
-            .Build()
-        .Build();
-
-    // Execute first query to get the required values to the client.
-    // Transaction control settings don't set CommitTx flag to keep transaction active
-    // after query execution.
-    auto result = session.ExecuteDataQuery(
-        query1,
-        TTxControl::BeginTx(TTxSettings::SerializableRW()),
-        std::move(params1)).GetValueSync();
-
-    if (!result.IsSuccess()) {
-        return result;
-    }
+void MultiStep(TQueryClient client, const TString& path) {
+    TMaybe<TResultSet> resultSet;
+    ThrowOnError(client.RetryQuerySync([&path, &resultSet](TSession session) {
+        ui64 seriesId = 2;
+        ui64 seasonId = 5;
+        auto query1 = Sprintf(R"(
+            --!syntax_v1
+            PRAGMA TablePathPrefix("%s");
+
+            DECLARE $seriesId AS Uint64;
+            DECLARE $seasonId AS Uint64;
+
+            SELECT first_aired AS from_date FROM seasons
+            WHERE series_id = $seriesId AND season_id = $seasonId;
+        )", path.c_str());
+
+        auto params1 = TParamsBuilder()
+            .AddParam("$seriesId")
+                .Uint64(seriesId)
+                .Build()
+            .AddParam("$seasonId")
+                .Uint64(seasonId)
+                .Build()
+            .Build();
+
+        // Execute first query to get the required values to the client.
+        // Transaction control settings don't set CommitTx flag to keep transaction active
+        // after query execution.
+        auto result = session.ExecuteQuery(
+            query1,
+            TTxControl::BeginTx(TTxSettings::SerializableRW()),
+            params1);
+
+        auto resultValue = result.GetValueSync();
+
+        if (!resultValue.IsSuccess()) {
+            return resultValue;
+        }
 ```
 
-To continue working within the current transaction, you need to get the current `transaction ID`:
+To continue working within the current transaction, you need to get the current `transaction id`:
 
 ```c++
-    // Get active transaction id
-    auto tx = result.GetTransaction();
+        // Get active transaction id
+        auto tx = resultValue.GetTransaction();
```

Comment:
As far as I understand, this doesn't actually obtain the transaction id. Something along the lines of `auto txId = resultValue.GetTransaction()->GetId();` would match the description better.